File: programmers/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    sort(participant.begin(),participant.end());
    sort(completion.begin(),completion.end());
    int i;
    for(i =0;i<completion.size();i++)
    {
        if( participant[i] != completion[i]){break;}
    }
    return participant[i];
}
/*string solution(vector<string> participant, vector<string> completion) {
    vector<string>::iterator it;
    for(int j =0;j<completion.size();j++)
    {
        bool ok = false;
         for(it=participant.begin();it<participant.end();it++){
            if(*it == completion[j]) {ok = true;participant.erase(it); break;}
        }
        if(!ok) {break;}
    }
    return participant[0];
}*/

File: JAVA/Programmers/완주하지 못한 선수.java
import java.util.*;
class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        int count[] = new int[participant.length];
        Hashtable<String, Integer> s = new Hashtable<>();
        for(int i = 0 ; i < completion.length;i++){
            if(s.get(completion[i])!= null){
                int cnt = s.get(completion[i]);
                cnt++;
                s.put(completion[i],cnt);
            }else{
                s.put(completion[i],1);
            }
        }
        for(int i = 0 ; i < participant.length; i++){
            if(s.get(participant[i]) == null){
                answer = participant[i];
            }else{
                int cnt = s.get(participant[i]);
                if(--cnt < 0){
                    answer = participant[i];
                }else{
                    s.put(participant[i],cnt);
                }
            }
        }
        return answer;
    }
}
